packages/caravan-fees/src/btcTransactionComponents.ts

```diff
@@ -0,0 +1,331 @@
+   * @param {Object} value - The witness UTXO
+   * @param {Buffer} value.script - The scriptPubKey of the output
+   * @param {number} value.value - The value of the output in satoshis
+   */
+  setWitnessUtxo(value: { script: Buffer; value: number }): void {
+    this._witnessUtxo = value;
+  }
+
+  /**
+   * Check if the input is valid
+   * @returns True if the amount is positive and exists, and txid and vout are valid
+   */
+  isValid(): boolean {
+    return this.hasAmount() && this._txid !== "" && this._vout >= 0;
+  }
+
+  /**
+   * Checks if the input has the required fields for PSBT creation.
+   */
+  hasRequiredFieldsforPSBT(): boolean {
+    return Boolean(
+      this._txid &&
+        this._vout >= 0 &&
+        (this._nonWitnessUtxo || this._witnessUtxo),
+    );
+  }
+
+  /**
+   * Converts the input template to a UTXO object.
+   */
+  toUTXO(): UTXO {
+    return {
+      txid: this._txid,
+      vout: this._vout,
+      value: this._amountSats.toString(),
+      prevTxHex: this._nonWitnessUtxo?.toString("hex"),
+      witnessUtxo: this._witnessUtxo,
+    };
+  }
+}
+/**
+ * Represents a Bitcoin transaction output
+ */
+export class BtcTxOutputTemplate extends BtcTxComponent {
+  private readonly _address: string;
+  private _malleable: boolean = true;
+
+  /**
+   * @param params - Output parameters
+   * @param params.address - Recipient address
+   * @param params.amountSats - Amount in satoshis  (as a string)
+   * @param params.locked - Whether the output is locked (immutable), defaults to false
+   * @throws Error if trying to create a locked output with zero amount
+   */
+  constructor(params: {
+    address: string;
+    amountSats?: Satoshis | undefined;
+    locked?: boolean;
+  }) {
+    super(params.amountSats || "0");
+    this._address = params.address;
+    this._malleable = !params.locked;
+
+    if (!this._malleable && this._amountSats.isEqualTo(0)) {
+      throw new Error("Locked outputs must have an amount specified.");
+    }
+  }
+
+  /** Get the recipient address */
+  get address(): string {
+    return this._address;
+  }
+
+  /** Check if the output is malleable (can be modified) */
+  get isMalleable(): boolean {
+    return this._malleable;
+  }
+
+  /**
+   * Set a new amount for the output
+   * @param amountSats - New amount in satoshis(as a string)
+   * @throws Error if trying to modify a non-malleable output
+   */
+  setAmount(amountSats: Satoshis): void {
+    if (!this._malleable) {
+      throw new Error("Cannot modify non-malleable output");
+    }
+    this._amountSats = new BigNumber(amountSats);
+  }
+
+  /**
+   * Add amount to the output
+   * @param amountSats - Amount to add in satoshis (as a string)
+   * @throws Error if trying to modify a non-malleable output
+   */
+  addAmount(amountSats: Satoshis): void {
+    if (!this._malleable) {
+      throw new Error("Cannot modify non-malleable output");
+    }
+    this._amountSats = this._amountSats.plus(new BigNumber(amountSats));
```

Comment:
This should be declared at the top of the class.

Reply:
can you please clarify what you meant by 
> This should be declared at the top of the class.

Reply:
Oh sorry! I meant with the other field declarations, but I forgot that this was inheriting from the base class which has it declared there. Sorry!